src/Components/Components/src/Rendering/ComponentState.cs

```diff
@@ -105,11 +105,23 @@ public bool TryDisposeInBatch(RenderBatchBuilder batchBuilder, out Exception exc
             return exception == null;
         }
 
+        // Callers expect this method to always return a faulted task.
```

Comment:
I think you're right. However it might be worth considering moving this `try`/`catch` block into `Renderer`'s `NotifyRenderCompleted` to make it clearer that the renderer is robust and not relying on assumptions about the error handling characteristics of `ComponentState`. That's just a nit really and I won't get bothered if you decide not to do it!